.github/workflows/build.yml

```
@@ -35,12 +35,20 @@ jobs:
       - name: Install deps
         uses: bahmutov/npm-install@v1
 
-      - name: Next.js cache
-        uses: actions/cache@v2
-        with:
-          path: ${{ github.workspace }}/.next/cache
-          key: ${{ runner.os }}-nextjs
+      # - name: Next.js cache
+      #   uses: actions/cache@v2
+      #   with:
+      #     path: ${{ github.workspace }}/.next/cache
+      #     key: ${{ runner.os }}-nextjs
 
-      - run: yarn prisma migrate deploy
+      - run: yarn prisma migrate dev --create-only
+      - name: Detect schema drift
+        run: |
+          if yarn prisma migrate status | grep -q 'Database schema is up to date'; then
+            echo "All OK"
+          else
+            echo "Missing migration files"
+            exit 1
+          fi;
```

This should work (tested locally)

```suggestion
          STATUS=$(yarn prisma migrate status)
          echo $STATUS

          if [[ $STATUS == *"Database schema is up to date"* ]]; then
            echo "All OK"
          else
            echo "Missing migration files"
            exit 1
          fi;
```

---

This code works already - the problem is that `yarn prisma migrate --create-only` creates an empty migration file even if there are no pending migrations so this will be a constant false negative.

---

I agree your bash scripting is prettier though. I hate bash and usually just bash it until it works 🤪